crates/uv-toolchain/src/discovery.rs

```diff
@@ -115,29 +116,12 @@ type ToolchainResult = Result<Toolchain, ToolchainNotFound>;
 
 /// The result of failed toolchain discovery.
 ///
-/// See [`InterpreterResult`].
+/// See [`ToolchainResult`].
 #[derive(Clone, Debug, Error)]
-pub enum ToolchainNotFound {
-    /// No Python installations were found.
-    NoPythonInstallation(ToolchainPreference, Option<VersionRequest>),
-    /// No Python installations with the requested version were found.
-    NoMatchingVersion(ToolchainPreference, VersionRequest),
-    /// No Python installations with the requested key were found.
-    NoMatchingKey(ToolchainPreference, PythonDownloadRequest),
-    /// No Python installations with the requested implementation name were found.
-    NoMatchingImplementation(ToolchainPreference, ImplementationName),
-    /// No Python installations with the requested implementation name and version were found.
-    NoMatchingImplementationVersion(ToolchainPreference, ImplementationName, VersionRequest),
-    /// The requested file path does not exist.
-    FileNotFound(PathBuf),
-    /// The requested directory path does not exist.
-    DirectoryNotFound(PathBuf),
-    /// No Python executables could be found in the requested directory.
-    ExecutableNotFoundInDirectory(PathBuf, PathBuf),
```

Comment:
We lose the ability to distinguish between these cases in error messages, but I don't think it's particularly critical.

Reply:
By "these cases", do you mean file vs. directory vs. executable?

Reply:
Specifically whether its a missing directory or a missing executable in the directory. We'll always raise the latter now.

Reply:
It would also be harder to encode things like "this exists but is not an executable" in the future. In practice, I don't think we were even using the variant right now.